hurumap/static/js/charts.js

```diff
@@ -0,0 +1,1359 @@
+        23: ["about one-quarter", "of"],
+        17: ["about one-fifth", "of"],
+        13: ["less than a fifth", "of"],
+        8: ["about 10 percent", "of"],
+        0: ["less than 10 percent", "of"],
+    }
+
+    chart.comparisonThresholds = d3.keys(chart.comparisonPhrases).map(Number);
+
+    chart.getComparisonThreshold = function(value) {
+        var threshold = chart.lastItem(chart.comparisonThresholds.filter(function(i) {
+            return i <= value
+        }));
+
+        return chart.comparisonPhrases[threshold]
+    }
+
+    chart.getComparisonNoun = function() {
+        if (chart.chartStatType == 'percentage' || chart.chartStatType == 'scaled-percentage') {
+            return 'rate';
+        } else if (chart.chartStatType == 'dollar') {
+            return 'amount';
+        }
+        return 'figure';
+    }
+
+    chart.trackEvent = function(category, action, label) {
+        // make sure we have Google Analytics function available
+        if (typeof(ga) == 'function') {
+            ga('send', 'event', category, action, label);
+        }
+    }
+
+    chart.generateURLs = function() {
+        if (chart.tableID) {
+            var geoIDs = chart.geoIDs;
+
+            chart.tableURL =        '/data/table/?table='+chart.tableID+'&primary_geo_id='+chart.primaryGeoID+'&geo_ids='+geoIDs.join(',');
+
+            // when showing distribution and maps, try to show relevant geos right from
+            // the start.
+            geoIDs = [];
+            if (chart.thisGeo.parent_geoid) {
+                // show our peers
+                var parentGeo = chart.parentGeo;
+                geoIDs.push(chart.thisGeo.geo_level + "|" + chart.thisGeo.parent_geoid);
+
+                if (parentGeo.parent_geoid) {
+                    geoIDs.push(parentGeo.geo_level + "|" + parentGeo.parent_geoid);
+                }
+            } else {
+                // show all children of this level
+                geoIDs.push(chart.thisGeo.child_level + "|" + chart.thisGeo.full_geoid);
+            }
+
+            chart.mapURL = '/data/map/?table='+chart.tableID+'&primary_geo_id='+chart.primaryGeoID+'&geo_ids='+geoIDs.join(',');
+            chart.distributionURL = '/data/distribution/?table='+chart.tableID+'&primary_geo_id='+chart.primaryGeoID+'&geo_ids='+geoIDs.join(',');
+        }
+    }
+
+    chart.download = function(){
+        var filter = function filter(node){
+            return (node.className !== 'action-links tool-group toggle-sub-group');
+        };
+        var bgcolor = '#fff';
+        domtoimage.toPng(chart.chartContainer.node(), { filter: filter, bgcolor: bgcolor })
+            .then(function (dataUrl) {
+                var link = document.createElement('a');
+                link.download = chart.chartChartTitle + '.png';
+                link.href = dataUrl;
+                link.click();
+            });
+        return false;
+    }
+
+    // ready, set, go
+    chart.init(options);
+    return chart;
+}
```

Comment:
@mtuchi Could you look at implementing the added feature in this JS instead as an "extension" of the Chart class?